queue/src/fiskaltrust.Middleware.Storage.Base/BaseStorageBootStrapper.cs

```diff
@@ -66,6 +68,44 @@ public StorageBaseInitConfiguration ParseStorageConfiguration(Dictionary<string,
             }
         }
 
+        public enum Migrations
+        {
+            JournalFRCopyPayload
+        }
+
+        public async Task PersistMigrationDataAsync(IJournalFRCopyPayloadRepository journalFRCopyPayloadRepository, IMiddlewareJournalFRRepository journalFRRepository)
+        {
+            if (await journalFRCopyPayloadRepository.HasEntriesAsync().ConfigureAwait(false) == false)
```

Comment:
Instead of checking if the table has any data and then performing the migrations we can do the following:

1. The `DatabaseMigrator` should return a `IEnumerable<Migrations>`.
2. In the `DatabaseMigrator` when applying migrations check if the migration `012_ftJournalFRCopyPayload.sqlite` was applied and if it was add it to the list of `Migrations` that are returned.
2. In the `SQLiteStorageBootstrapper` call the method `PerformMigrationInitialization` with the `IEnumerable<Migrations>` we get from the `DatabaseMigrator`.

Reply:
I've implemented your suggestions in last commit and also described it in the commit message what I already did :)